src/hotspot/share/runtime/signature.hpp

```diff
@@ -339,10 +339,13 @@ class Fingerprinter: public SignatureIterator {
   void do_type_calling_convention(BasicType type);
 
   friend class SignatureIterator;  // so do_parameters_on can call do_type
-  ATTRIBUTE_NO_UBSAN
+
   void do_type(BasicType type) {
     assert(fp_is_valid_type(type), "bad parameter type");
-    _accumulator |= ((fingerprint_t)type << _shift_count);
+    if (_param_count <= fp_max_size_of_parameters) {
+      _accumulator |= ((fingerprint_t)type << _shift_count);
+    }
+    _param_count++;
```

Comment:
Let's not introduce another field here.  We already have _shift_count that we can use to detect overflow.